docs/kubernetes/user-guides/snapshots.md

```diff
@@ -122,3 +125,150 @@
+     PDs; what is below matches the first example in this document.
+     ```console
+     kubectl apply -f - <<EOF
+       kind: PersistentVolumeClaim
+       apiVersion: v1
+       metadata:
+         name: restored-pvc
+       spec:
+         accessModes:
+           - ReadWriteOnce
+         storageClassName: csi-gce-pd
+         resources:
+           requests:
+             storage: 6Gi
+         dataSource:
+           kind: VolumeSnapshot
+           name: restored-snapshot
+           apiGroup: snapshot.storage.k8s.io
+     EOF
+     ```
+  1. Finally, create a pod referring to the `PersistentVolumeClaim`. The PD CSI
+     driver will provision a `PersistentVolume` and populate it from the
+     snapshot. The pod from `examples/kubernetes/snapshot/restored-pod.yaml` is
+     set to use the PVC created above. After `kubectl apply`'ing the pod, run
+     `kubectl exec restored-pod -- ls -l /demo/data/` to confirm that the
+     snapshot has been restored correctly.
+
+#### Troubleshooting
+
+If the `VolumeSnapshot`, `VolumeSnapshotContents` and `PersistentVolumeClaim`
+are not all mutually syncrhonized, the pod will not start. Try `kubectl describe
+volumesnapshotcontent restored-snapshot-content` to see any error messages
+relating to the binding together of the `VolumeSnapshot` and the
+`VolumeSnapshotContents`. Further errors may be found in the snapshot controller
+logs:
+
+```console
+kubectl logs snapshot-controller-0 | tail
+```
+
+Any errors in `kubectl describe pvc restored-pvc` may also shed light on any
+troubles. Sometimes there is quite a wait to dynamically provision the
+`PersistentVolume` for the PVC; the following command will give more
+information.
+
+```console
+kubectl logs -n gce-pd-csi-driver csi-gce-pd-controller-0 -c csi-provisioner | tail
+```
+
+### Tips on Migrating from Alpha Snapshots
+
+The api version has changed between the alpha and beta releases of the CSI
+snapshotter. In many cases, snapshots created with the alpha will not be able to
+be used with a beta driver. In some cases, they can be migrated.
+
+*You may lose all data in your snapshot!*
+
+*These instructions not work in your case and could delete all data in your
+snapshot!*
+
+These instructions happened to work with a particular GKE configuration. Because
+of the variety of alpha deployments, it is not possible to verify these steps
+will work all of the time. The following has been tested with a GKE cluster of
+master version 1.17.5-gke.0 using the following image versions for the
+snapshot-controller and PD CSI driver:
+
+  * snapshot-controller:v2.0.1
+  * csi-provisioner:v1.5.0-gke.0
+  * gke.gcr.io/csi-attacher:v2.1.1-gke.0
+  * gke.gcr.io/csi-resizer:v0.4.0-gke.0
+  * gke.gcr.io/csi-snapshotter:v2.1.1-gke.0
+  * gke.gcr.io/gcp-compute-persistent-disk-csi-driver:v0.7.0-gke.0
+
+#### Migrating by Restoring from a Manually Provisioned Snapshot
+
+The idea is to provision a `VolumeSnapshotContents` in your new or upgraded beta
```

Comment:
clarify this is a beta VolumeSnapshotContent

Reply:
done